src/components/CellEmphasis.test.js

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import Adapter from 'enzyme-adapter-react-16';
+import { shallow, configure } from 'enzyme';
+import expect from 'expect';
+import CellEmphasis from './CellEmphasis';
+
+configure({ adapter: new Adapter() });
+
+const fakeHoveredCellInfo = {
+  cellId: '123',
+  mappings: { xy: [1, 1], tsne: [2, 2] },
+  uuid: 1,
+  status: 'subcluster: Hippocampus; cluster: Excitatory neurons',
+};
+
+const fakeViewInfoInside = {
+  width: 10,
+  height: 10,
+  viewport: {
+    project() {
+      return [5, 5];
+    },
+  },
+};
+
+const fakeViewInfoOutsideAbove = {
+  width: 10,
+  height: 10,
+  viewport: {
+    project() {
+      return [0, -1];
+    },
+  },
+};
+
+const fakeViewInfoOutsideBelow = {
+  width: 10,
+  height: 10,
+  viewport: {
+    project() {
+      return [0, 11];
+    },
+  },
+};
+
+const fakeViewInfoOutsideLeft = {
+  width: 10,
+  height: 10,
+  viewport: {
+    project() {
+      return [-1, 0];
+    },
+  },
+};
+
+const fakeViewInfoOutsideRight = {
+  width: 10,
+  height: 10,
+  viewport: {
+    project() {
+      return [11, 0];
+    },
+  },
+};
```

Comment:
You might make a util function ... though it is a trade-off: in some cases having the static examples will be more clear than having some complicated function that makes examples. Your call.
```javascript
function makeFakeViewInfo(x, y) {
  return {
    width: 10,
    height: 10,
    viewport: {
      project() {
        return [x, y];
      },
    },
  };
}
```

Reply:
I was wondering this when I wrote the test. I think in this case the function is simple enough that it is reasonable to use so I have updated.